superset/utils/core.py

```diff
@@ -1097,11 +1107,9 @@ def merge_extra_form_data(form_data: Dict[str, Any]) -> None:
         {"isExtra": True, **fltr} for fltr in append_adhoc_filters  # type: ignore
     )
     if append_filters:
-        adhoc_filters.extend(
-            simple_filter_to_adhoc({"isExtra": True, **fltr})  # type: ignore
-            for fltr in append_filters
-            if fltr
-        )
+        for key, value in form_data.items():
+            if re.match("adhoc_filter.*", key):
+                append_filters_to_adhoc_filters(value, append_filters)
```

Comment:
To make this logic seem more straightforward(Which iteme are we going to update), I suggest this:

```suggestion
append_filters.extend(form_data.get('adhoc_filter', []))
append_filters.extend(form_data.get('adhoc_filters', []))
append_filters = [simple_filter_to_adhoc({"isExtra": True, **f}) for f in append_filters if f]
```

Reply:
The point of iterating through form_data keys and checking regex is to make it more generic and future proof - it should work with any control that's name is like `adhoc_filter.*`, so not only `adhoc_filter` and `adhoc_filter_b`

Reply:
Okay! Thanks for the explanation.